### PR TITLE
Issue 7 tests

### PR DESCRIPTION
**Pull request by skyerus**
Created to address [issue 7](https://github.com/open-feature/golang-sdk/issues/7)

### CODE REVIEW

**Comment by skyerus** [comment]
Included https://github.com/golang/mock as a dependency to generate mocks of our interfaces for testing purposes. If including this is an issue we can perhaps just generate the mocks and not import it?

File: pkg/openfeature/provider.go
Comment (toddbaert) [inline pkg/openfeature/provider.go:60]: Might want to change this comment to `Metadata provides provider name`

File: pkg/openfeature/client_test.go
Comment (toddbaert) [inline pkg/openfeature/client_test.go:69]: :+1:

File: pkg/openfeature/client_test.go
Comment (toddbaert) [inline pkg/openfeature/client_test.go:141]: It MAY be worth testing that in the case of abnormal execution of the provider, the default is returned as well as an error, but I don't feel too strongly about it.

File: pkg/openfeature/client_test.go
Comment (toddbaert) [inline pkg/openfeature/client_test.go:145]: I don't think we have to much worry about this one because of goroutines.

**Review by toddbaert — APPROVED** [review]
This looks good to me, thanks! Personally I would recommend waiting for an approval from @davejohnston or @thomaspoignant as well.

File: pkg/openfeature/provider.go
Comment (skyerus) [reply pkg/openfeature/provider.go]: Done

File: pkg/openfeature/client_test.go
Comment (skyerus) [reply pkg/openfeature/client_test.go]: Done

File: pkg/openfeature/client_test.go
Comment (skyerus) [reply pkg/openfeature/client_test.go]: Updated

**Pull request merged**
